src/main/groovy/com/bmuschko/gradle/vagrant/tasks/Vagrant.groovy

```diff
@@ -37,7 +41,8 @@ class Vagrant extends DefaultTask {
     /**
      * The directory the targeted Vagrant box resides in.
      */
-    @Input
+    @PathSensitive(PathSensitivity.RELATIVE)
+    @InputDirectory
     File boxDir
```

Comment:
I'm not sure this is really a good idea since the default value for `boxDir` is `projectDir`. It would probably be better deprecate this and add a new property (possibly of type `DirectoryProperty`) with a default value of `${projectDir}/vagrant`. Otherwise this code will effectively add all sorts of things like the buildDir, gradle wrapper files and temporary stuff in `.gradle` to the inputs of this task.

Reply:
So in other words, users will always have to reconfigure this to get decent up to date tracking and caching.

Reply:
Changing to the new `Provider` types will be quite a bit of work and should be done across the board for all custom task types to be consistent. Pointing it to a different default value could be a good idea. I'd be OK with that if you want make the change. Kind of a breaking change but that's fine.

Reply:
For now I've only changed the default value to point to `project.file("vagrant")`. I think updating everything at once to Providers API makes sense but warrants a major version bump.